src/toolchain/distributable.rs

```diff
@@ -115,6 +115,14 @@ impl<'a> DistributableToolchain<'a> {
         Ok(())
     }
 
+    pub(crate) fn components(&self) -> anyhow::Result<Vec<crate::dist::manifest::ComponentStatus>> {
+        let manifestation = self.get_manifestation()?;
+        let config = manifestation.read_config()?.unwrap_or_default();
+        let manifest = self.get_manifest()?;
+        let components = manifest.query_components(self.desc(), &config)?;
+        Ok(components)
```

Comment:
Style nit: I would avoid binding `components` here, in favor of just yielding `manifest.query_components(self.desc(), &config)`.

Reply:
Done.